PWGHF: add selection on generated PV z position in creators

fgrosa opened the pull request:
This PR implements the selection on the generated PV z position. In order to avoid human mistakes, the same value used in the reconstructed part is retrieved by the workflow of the reconstructed part. Not yet implemented for the XiPi creator because no event selection is present for the XiPi creator for the reconstructed part (@ZFederica let me know if you want me to include it).

mfaggin reviewed (approved):
thanks @fgrosa

vkucera commented on PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx:
```diff
@@ -385,8 +386,22 @@ struct HfCandidateSigmac0plusplusMc {
+  void init(InitContext& initContext)
+  {
+    auto& workflows = initContext.services().get<RunningWorkflowInfo const>();
+    for (const DeviceSpec& device : workflows.devices) {
+      if (device.name.compare("hf-candidate-creator-3prong") == 0) { // here we assume that the hf-candidate-creator-3prong is in the workflow
+        for (const auto& option : device.options) {
+          if (option.name.compare("zPvPosMax") == 0) {
+            zPvPosMax = option.defaultValue.get<float>();
+          }
+        }
+      }
```
Add `break`s.

vkucera commented on PWGHF/TableProducer/candidateCreator2Prong.cxx:
```diff
@@ -665,11 +666,27 @@ struct HfCandidateCreator2ProngExpressions {
+  // inspect for which zPvPosMax cut was set for reconstructed
+  void init(InitContext& initContext)
+  {
+    auto& workflows = initContext.services().get<RunningWorkflowInfo const>();
```
```suggestion
    const auto& workflows = initContext.services().get<RunningWorkflowInfo const>();
```

fgrosa commented:
Hi @vkucera I implemented your comments in [c8f9cbf](https://github.com/AliceO2Group/O2Physics/pull/5839/commits/c8f9cbf79a566baffa8c31a1d94e39dbe49ad1ff)

vkucera commented:
> Hi @vkucera I implemented your comments in [c8f9cbf](https://github.com/AliceO2Group/O2Physics/pull/5839/commits/c8f9cbf79a566baffa8c31a1d94e39dbe49ad1ff)

Thanks @fgrosa , some are still missing for the `if` blocks of the inner `for` loops.

vkucera commented on PWGHF/TableProducer/candidateCreator2Prong.cxx:
```diff
@@ -671,14 +671,16 @@ struct HfCandidateCreator2ProngExpressions {
           if (option.name.compare("zPvPosMax") == 0) {
             zPvPosMax = option.defaultValue.get<float>();
           }
+          break;
```
This should be inside the `if` block.

fgrosa commented:
> > Hi @vkucera I implemented your comments in [c8f9cbf](https://github.com/AliceO2Group/O2Physics/pull/5839/commits/c8f9cbf79a566baffa8c31a1d94e39dbe49ad1ff)
> 
> Thanks @fgrosa , some are still missing for the `if` blocks of the inner `for` loops.

Thanks for spotting it, it should be now fixed.

ZFederica commented:
Hi @fgrosa, if you don't mind, I think it would be a useful implementation also in the candidateCreatorToXiPi, thanks!

fgrosa commented:
> Hi @fgrosa, if you don't mind, I think it would be a useful implementation also in the candidateCreatorToXiPi, thanks!

I will do it in a separated PR not to delay this one (I am now following the PB).

alibuild reviewed (approved):
Auto-approving on behalf of @fgrosa.
